velox/exec/Spill.cpp

```diff
@@ -175,6 +182,28 @@ uint64_t SpillState::appendToPartition(
   return partitionWriter(id)->write(rows, folly::Range<IndexRange*>(&range, 1));
 }
 
+uint64_t SpillState::appendToPartition(
+    const SpillPartitionId& id,
+    const SpillRows& rows,
+    RowContainer* rowContainer) {
+  VELOX_CHECK(
+      isPartitionSpilled(id), "Partition {} is not spilled", id.toString());
+
+  TestValue::adjust(
+      "facebook::velox::exec::SpillState::appendToPartition", this);
+
+  initPartitionWriter(id);
+
+  uint64_t bytes = 0;
+  for (const auto& row : rows) {
```

Comment:
Here you have got the exactly row size, so you don't need an estimated initial size 1'000
```
batch_->createStreamTree(
          std::static_pointer_cast<const RowType>(type_),
          1'000,
          serdeOptions_.get());
```